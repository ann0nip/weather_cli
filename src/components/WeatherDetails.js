import React from "react";
import moment from "moment";
import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
  CardContent,
  CardMedia,
  Typography,
  Divider,
  List,
  ListItem,
  ListItemText,
} from "@material-ui/core";

const useStyles = makeStyles({
  root: {
    boxShadow: "none",
    marginTop: 10,
    backgroundColor: "#7db4fc",
  },
  title: {
    fontSize: 14,
  },
  media: {
    height: 55,
    backgroundSize: "auto",
  },
  forecastIcon: {
    height: 55,
    width: 55,
    backgroundSize: 55,
  },
  divider: {
    margin: 10,
  },
});

export default function WeatherDetails(props) {
  const classes = useStyles();
  const {
    current: { city, weather, more },
    forecast,
  } = props.data;

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="body1" display={"inline"}>
          {city},
        </Typography>
        <span> </span>
        <Typography variant={"body1"} display={"inline"}>
          {moment(new Date()).format("ll")}
        </Typography>
        <br></br>
        <Typography variant={"h6"} display={"inline"}>
          {weather[0].main},
        </Typography>
        <span> </span>
        <Typography variant={"h6"} display={"inline"}>
          {Math.trunc(more.temp)} Cº
        </Typography>
        <CardMedia
          className={classes.media}
          image={`http://openweathermap.org/img/wn/${weather[0].icon}@2x.png`}
          title={weather[0].description}
        />
        <List>
          {forecast.map((el, k) => {
            return (
              <React.Fragment key={k}>
                <ListItem>
                  <ListItemText primary={moment(el.dt_txt).format("MMM Do")} />
                  <ListItemText
                    primary={`${Math.trunc(el.main.temp_max)} Cº`}
                  />
                  <CardMedia
                    className={classes.forecastIcon}
                    image={`http://openweathermap.org/img/wn/${el.weather[0].icon}@2x.png`}
                    title={el.weather[0].description}
                  />
                </ListItem>
                <Divider variant="middle" />
              </React.Fragment>
            );
          })}
        </List>
      </CardContent>
    </Card>
  );
}
