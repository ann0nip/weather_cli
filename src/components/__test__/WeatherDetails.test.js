import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { data } from "./WeatherDetails.mock";

import WeatherDetails from "../WeatherDetails";

describe("<WeatherDetails />", () => {
  it("render <WeatherDetails /> component", () => {
    const tree = renderer.create(<WeatherDetails data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
