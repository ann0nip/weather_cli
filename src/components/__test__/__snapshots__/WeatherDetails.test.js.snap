// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDetails /> render <WeatherDetails /> component 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-displayInline"
    >
      Buenos Aires
      ,
    </p>
    <span>
       
    </span>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-displayInline"
    >
      Jun 29, 2020
    </p>
    <br />
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-displayInline"
    >
      Clouds
      ,
    </h6>
    <span>
       
    </span>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-displayInline"
    >
      10
       Cº
    </h6>
    <div
      className="MuiCardMedia-root makeStyles-media-3"
      style={
        Object {
          "backgroundImage": "url(\\"http://openweathermap.org/img/wn/04n@2x.png\\")",
        }
      }
      title="broken clouds"
    />
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Jun 30th
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            11 Cº
          </span>
        </div>
        <div
          className="MuiCardMedia-root makeStyles-forecastIcon-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://openweathermap.org/img/wn/10n@2x.png\\")",
            }
          }
          title="moderate rain"
        />
      </li>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Jul 1st
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            10 Cº
          </span>
        </div>
        <div
          className="MuiCardMedia-root makeStyles-forecastIcon-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://openweathermap.org/img/wn/10n@2x.png\\")",
            }
          }
          title="moderate rain"
        />
      </li>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Jul 2nd
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            8 Cº
          </span>
        </div>
        <div
          className="MuiCardMedia-root makeStyles-forecastIcon-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://openweathermap.org/img/wn/04n@2x.png\\")",
            }
          }
          title="broken clouds"
        />
      </li>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Jul 3rd
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            9 Cº
          </span>
        </div>
        <div
          className="MuiCardMedia-root makeStyles-forecastIcon-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://openweathermap.org/img/wn/04n@2x.png\\")",
            }
          }
          title="overcast clouds"
        />
      </li>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Jul 4th
          </span>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            10 Cº
          </span>
        </div>
        <div
          className="MuiCardMedia-root makeStyles-forecastIcon-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://openweathermap.org/img/wn/04n@2x.png\\")",
            }
          }
          title="overcast clouds"
        />
      </li>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
    </ul>
  </div>
</div>
`;
